libr/core/canal.c

```diff
@@ -860,6 +859,12 @@ static int core_anal_fcn(RCore *core, ut64 at, ut64 from, int reftype, int depth
 		}
 		free (next);
 	}
+	if (!strcmp (R_SYS_ARCH, "x86")) {
```

Comment:
thats not correct, you should do: !strcmp(core->anal->cur->arch, "x86") (check if core->anal->cur is non null before doing it)